build/profiles/freenas/packages/base-os/config

```diff
@@ -53,6 +53,11 @@ post-upgrade =
 	   done
 	fi
 
+    rm -rf /usr/local/www/freenasUI
+
+    touch /data/need-update
```

Comment:
Well, that is the reason!

Any reason why? What about when we upload config from previous version, then middleware starts over config of previous version of FreeNAS and bad things can happen?

Reply:
Yes, that could potentially cause middlewared fail to start causing at least 120 seconds boot delay.

I dont remember details why it was done this way, i remember something about avoiding double reboots, potentially because the autotune, because it tries to reboot to because of loader.conf changes.

Reply:
@themylogin what I meant we could do here, is that we could adjust ix-update to run very early in the boot process, before middlewared and see if anything falls apart

Reply:
`ix-update` does `midclt call etc.generate loader > /dev/null` so it needs middleware running, I guess we leave all this as it is as you have initially suggested?

Reply:
Up to you